### DISCUSSION TITLE
Extrapolate Encoder from pre-trained 3D net to classify

### DISCUSSION THREAD

**Discussion by robsver** [Project-MONAI/MONAI#7211] [Q&A]
Hello everyone, 
I would like to know if it is possible to extrapolate from a **pre-trained 3D model for segmentation** (_e.g.: UNETR_) its encoder part and use it to perform **classification**. 

How would you do this?

**Comment by KumoLiu** [comment]
Hi @robsver, you could add a hook in the constructor, a simple demo here.
```
        self.extra_outputs = {}
        def forward_hook(layer_name):
            def hook(module, input, output):
                self.extra_outputs[layer_name] = output

            return hook

        self.encoder4.register_forward_hook(forward_hook("layer4"))
```
Hope it helps, thanks!

**Reply by robsver** [reply]
Hi @KumoLiu, Thanks a lot.
